lib/client/client.go

```diff
@@ -186,8 +186,22 @@ func (proxy *ProxyClient) ReissueUserCerts(ctx context.Context, params ReissuePa
 		return trace.Wrap(err)
 	}
 
+	localKeyAgent := proxy.localAgent()
+
+	// The re-issued certificate may contain new roles that change our
+	// permissions with *any* service (e.g. Kubernetes, Database) in the
+	// cluster. We want to make sure no cached certificates are preserved
+	// once we have the re-issued certificate, forcing each service to get
+	// a new certificate (with the potentially updated roles) on first
+	// access, so we explicitly delete them all.
+	log.Debugf("Deleting all user certs...")
+	if err = localKeyAgent.DeleteUserCerts(params.RouteToCluster, WithAllCerts...); err != nil {
```

Comment:
Wouldn't this effectively prevent more than a single user/service cert to be stored per cluster at a time? It seems this deletion should occur only if the list of active requests differs from the one used with the stored key.

Also, given the list of active requests should be obtainable from the stored key itself, this logic could be more appropriately moved directly into `LocalKeyAgent.AddKey`.

Reply:
+1, existing cert deletion should happen only when requesting additional roles, not on every reissue
please move this to `/tool/tsh/`, right after a successful access request approval

Reply:
Moved.